fhd_core/source_modeling/fhd_diffuse_model.pro

```diff
@@ -8,51 +8,71 @@ astr=obs.astr
+freq_center=obs.freq_center
 IF Keyword_Set(skymodel) THEN diffuse_spectral_index=skymodel.diffuse_spectral_index ELSE diffuse_spectral_index=0
 apply_astrometry, obs, x_arr=meshgrid(dimension,elements,1), y_arr=meshgrid(dimension,elements,2), ra_arr=ra_arr, dec_arr=dec_arr, /xy2ad
 radec_i=where(Finite(ra_arr))
 IF Keyword_Set(flatten_spectrum) THEN alpha_corr=obs.alpha ELSE alpha_corr=0.
 IF N_Elements(diffuse_units_kelvin) EQ 0 THEN diffuse_units_kelvin = 0
 IF Keyword_Set(diffuse_units_kelvin) THEN from_jy_per_sr = 0 ELSE from_jy_per_sr = 1
 
-;freq_use=where((*obs.baseline_info).freq_use,nf_use)
-;f_bin=(*obs.baseline_info).fbin_i
-;fb_use=Uniq(f_bin[freq_use])
-;nbin=N_Elements(fb_use)
-;freq_arr=(*obs.baseline_info).freq
-;alpha=obs.alpha
-;freq_norm=freq_arr^(-alpha)
-;;freq_norm/=Sqrt(Mean(freq_norm^2.))
-;freq_norm/=Mean(freq_norm) 
-;d_freq=Median(Float(deriv(freq_arr)))
-;
-;freq_norm=freq_norm[freq_use[fb_use]]
-;
-;freq_arr_use=freq_arr[freq_use[fb_use]]/1E6
-;fb_hist=histogram(f_bin[freq_use],min=0,bin=1)
-;nf_arr=fb_hist[f_bin[freq_use[fb_use]]]
-
 IF file_test(model_filepath) EQ 0 THEN RETURN,Ptrarr(n_pol)
-var_dummy=getvar_savefile(model_filepath,names=var_names)
-;save file must have one variable called 'hpx_inds', one called 'nside', and at least one other variable. If there are multiple other variables, it must be called 'model_arr'
-hpx_inds=getvar_savefile(model_filepath,'hpx_inds')
-nside=getvar_savefile(model_filepath,'nside')
-var_name_inds=where((StrLowCase(var_names) NE 'hpx_inds') AND (StrLowCase(var_names) NE 'nside'))
-var_names=var_names[var_name_inds]
-var_name_use=var_names[(where(StrLowCase(var_names) EQ 'model_arr',n_match))[0]>0] ;will pick 'model_arr' if present, or the first variable that is not 'hpx_inds' or 'nside'
-model_hpx_arr=getvar_savefile(model_filepath,var_name_use)
 
-model_spectra_i=where(StrLowCase(var_names) EQ 'model_spectral_arr',n_match)
-IF n_match GE 1 THEN diffuse_spectral_index=getvar_savefile(model_filepath,var_names[model_spectra_i])
-IF n_spectral LE 0 THEN undefine_fhd,diffuse_spectral_index
+upname=StrUpCase(model_filepath)
+skyh5_check=strpos(upname,'.SKYH5')
+IF skyh5_check NE -1 THEN BEGIN ;read skyh5 file
+    file_id = H5F_OPEN(model_filepath)
+    model_hpx_arr = H5D_READ(H5D_OPEN(file_id, '/Data/stokes'))
+    hpx_inds = H5D_READ(H5D_OPEN(file_id, '/Header/hpx_inds'))
+    hpx_order = H5D_READ(H5D_OPEN(file_id, '/Header/hpx_order'))
+    nside = H5D_READ(H5D_OPEN(file_id, '/Header/nside'))
+    nfreqs = H5D_READ(H5D_OPEN(file_id, '/Header/Nfreqs'))
+    npixels = H5D_READ(H5D_OPEN(file_id, '/Header/Ncomponents'))
+    if nfreqs eq 1 then model_hpx_arr = reform(model_hpx_arr) else begin
+        freq_arr = H5D_READ(H5D_OPEN(file_id, '/Header/freq_array'))
+        null = min(freq_arr-freq_center, min_ind, /absolute)
+        model_hpx_arr = reform(model_hpx_arr[*, min_ind, *])
+    endelse
+    if StrUpCase(hpx_order) eq 'NESTED' then begin ;reorder to ring ordering
```

Comment:
Can't you use a library function (such as https://healpix.jpl.nasa.gov/html/idlnode51.htm) to convert from nested to ring index ordering?

Reply:
This uses the library function, but unfortunately that function requires implicit Healpix indexing (rather than allowing pixel indices to be provided as a separate list). That makes this reordering a little convoluted.